home-assistant-plugin/custom_components/xknx/sensor.py

```diff
@@ -103,6 +105,20 @@ def unit_of_measurement(self):
         """Return the unit this state is expressed in."""
         return self.device.unit_of_measurement()
 
+    @property
+    def device_class(self):
+        """Return the device class of the sensor."""
+        if self.device.sensor_value.value_type == "humidity":
+            return DEVICE_CLASS_HUMIDITY
+        if self.device.sensor_value.value_type in \
+                ("illuminance", 'luminous_flux'):
+            return DEVICE_CLASS_ILLUMINANCE
+        if self.device.sensor_value.value_type == "pressure":
+            return DEVICE_CLASS_PRESSURE
+        if self.device.sensor_value.value_type == "temperature":
+            return DEVICE_CLASS_TEMPERATURE
+        return None
+
```

Comment:
device_class for some value types. #194 
Since it is only relevant for home assistant I put it here. If we defined it in the corresponding DPTArray subclasses we would have to use strings or enums and match the HA constant here.